test/all.test.ts

```diff
@@ -0,0 +1,52 @@
+import test from 'ava';
+
+import * as promiseUtils from '../src/index';
+
+test('throws errors in order with multiple errors', async t => {
```

Comment:
this is a little different than Promise.all (which would throw the first error that occurs, not the first in the array). this doesn't seem unreasonable, but let's make sure we call it out in the documentation